### ISSUE TITLE
SQL bug: update_term_taxonomy_counts() clears all term_counts when passed a term_taxonomy_id

### ISSUE THREAD

**Issue by youri--** [eventespresso/event-espresso-core#431]
Using EE version 4.9.61.p.

**Description of bug**
When using custom code to assign an event category to an event, all term counts are reset to 0, except for the term_taxonomy that is the category assigned, which gets the count of all term counts combined. See "Additional context" for details and proposed change.

**Expected behavior**
Term count should be updated for the $term_taxonomy_id passed.

**To Reproduce**
Use an EE_Event object and add an existing event category to it.

```
$new_event = EE_Event::new_instance( array(
	'EVT_name' =>  'Event Name',
	'EVT_desc' =>'Some text',
	'EVT_display_ticket_selector' => 0,
	'EVT_external_URL' => 'http://google.com/',
) );
$new_post_id = $new_event->save();

$new_event->add_event_category( 'my-existing-event-category-slug' );
```

This will call in turn call method `add_event_category()` in EEM_CPT_Base.model.php and from there eventually we'll end up in method `update_term_taxonomy_counts()` where the SQL is problematic when a term_taxonomy_id was passed.

**Stack**
```
EEM_Term_Relationship.model.php:174, EEM_Term_Relationship->insert()
EE_HABTM_Relation.php:169, EE_HABTM_Relation->add_relation_to()
EEM_Base.model.php:2562, EEM_Event->add_relationship_to()
EEM_Event.model.php:710, EEM_Event->add_relationship_to()
EEM_CPT_Base.model.php:331, EEM_Event->add_event_category()
EE_CPT_Base.class.php:125, EE_Event->add_event_category()
taf.php:1405, taf::post_form_shortcode()
shortcodes.php:319, do_shortcode_tag()
shortcodes.php:197, preg_replace_callback()
shortcodes.php:197, do_shortcode()
class-wp-hook.php:286, WP_Hook->apply_filters()
plugin.php:203, apply_filters()
post-template.php:240, the_content()
page.php:169, include()
template-loader.php:74, require_once()
wp-blog-header.php:19, require()
index.php:17, {main}()
```

**Additional context**
in the file `event-espresso-core-reg/core/db_models/EEM_Term_Relationship.model.php` starting on line 129, we find the following method:
```
    /**
     * Makes sure all term-taxonomy counts are correct
     *
     * @param int   $term_taxonomy_id the id of the term taxonomy to update. If NULL, updates ALL
     * @global wpdb $wpdb
     * @return int the number of rows affected
     */
    public function update_term_taxonomy_counts($term_taxonomy_id = null)
    {
        //because this uses a subquery and sometimes assigning to column to be another column's
        //value, we just write the SQL directly.
        global $wpdb;
        if ($term_taxonomy_id) {
            $second_operand = $wpdb->prepare('%d', $term_taxonomy_id);
        } else {
            $second_operand = 'tr.term_taxonomy_id';
        }
        $rows_affected = $this->_do_wpdb_query(
            'query',
            array(
                "
                UPDATE {$wpdb->term_taxonomy} AS tt 
                SET count = (
                    select count(*) as proper_count from {$wpdb->term_relationships} AS tr 
                    WHERE tt.term_taxonomy_id = $second_operand
                )",
            )
        );
        return $rows_affected;
    }
```

The SQL query is fine when $term_taxonomy_id = null (no parameter was passed). This gives this SQL statement:
```
UPDATE wp_term_taxonomy AS tt 
SET count = (
    select count(*) as proper_count from wp_term_relationships AS tr 
    WHERE tt.term_taxonomy_id = tr.term_taxonomy_id
);
```
and results in all records being updated to their proper relationship count.

However, when a $term_taxonomy_id is passed (5771 in this example), we get the following SQL statement being executed:
```
UPDATE wp_term_taxonomy AS tt 
SET count = (
    select count(*) as proper_count from wp_term_relationships AS tr 
    WHERE tt.term_taxonomy_id = 5771
);
```
This results in all wp_term_taxonomy records having their `count` reset to 0, except for the record with the matching term_taxonomy_id, 5771 in this case, which will be set to the total number of all relationships of all terms combined.

The WHERE clause references the wrong table in this case, and there's an additional WHERE clause missing. The SQL statement (again '5771' is passed as $term_taxonomy_id as an example), should IMHO be:
```
UPDATE wp_term_taxonomy AS tt 
SET count = (
    select count(*) as proper_count from wp_term_relationships AS tr 
    WHERE tr.term_taxonomy_id = 5771
)
WHERE tt.term_taxonomy_id = 5771;
```
Which would result in the term count for this specific term_taxonomy being updated to its actual count.

**Environment:**
 - Version of WordPress: 4.9.5
 - PHP Version 7.1

_edit: in the last sql block, moved the semicolon to the end of the statement where it should've been_

**Comment by mnelson4** [comment]
thanks for the detailed bug report @youri--. I'll try to get to this ASAP, but if you'd like to submit a pull request with your suggested fix and some unit tests (probably to `EEM_Term_Relationship_Test`) that would move things along quicker

**Comment by mnelson4** [comment]
this issue pull request was merged into master during 4.9.62 so moving this to that milestone
